src/Updater.php

```diff
@@ -177,20 +172,64 @@ public function executeUpdate($module, $update_definition_name) {
         $new_config = NestedArray::mergeDeep($new_config, $update_actions['add']);
       }
 
-      if ($this->updateConfig($configName, $new_config, $expected_config, $delete_keys)) {
+      if ($this->updateConfig($configName, $new_config, $configChange['expected_config'], $delete_keys)) {
         $this->logger->info($this->t('Configuration @configName has been successfully updated.', ['@configName' => $configName]));
       }
       else {
-        $successful = FALSE;
+        $this->successfulExecute = FALSE;
         $this->logger->warning($this->t('Unable to update configuration for @configName.', ['@configName' => $configName]));
       }
     }
+  }
 
-    // Dispatch event after update has finished.
-    $event = new ConfigurationUpdateEvent($module, $update_definition_name, $successful);
-    $this->eventDispatcher->dispatch(UpdateHelperEvents::CONFIGURATION_UPDATE, $event);
+  /**
+   * Installs modules.
+   *
+   * @param array $modules
+   *   List of module names.
+   */
+  protected function installModules(array $modules) {
+    foreach ($modules as $module) {
+      try {
+        if ($this->moduleInstaller->install([$module])) {
+          $this->logger->info($this->t('Module @module is successfully enabled.', ['@module' => $module]));
+        }
+        else {
+          $this->logger->warning($this->t('Unable to enable @module.', ['@module' => $module]));
+          $this->successfulExecute = FALSE;
```

Comment:
This pattern of logging a warning and setting an error makes me think this could be a helper method on the class.